### PR TITLE
[JS-loaders] Conceptual docs

### PR DESCRIPTION
**Pull request by dmceachernmsft**
# What

Add SB page for the loader functions
# Why

Provides setup and guidance for contoso to use the loader functions

https://skype.visualstudio.com/SPOOL/_workitems/edit/3880787
https://skype.visualstudio.com/SPOOL/_workitems/edit/3880790
# How Tested

Validated locally check SB8 link

### CODE REVIEW

File: packages/storybook8/stories/Composites/TypeScriptLoading/Docs.mdx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+Since all different [Node](https://nodejs.org/en) frameworks are just JavaScript, CSS, and HTML you can use each of the
+different frameworks inside each other. In the case of using [React](https://react.dev/) inside your application you would use
+the `createRoot` function to attach a React node to your application and then use your react component inside that.
+
+Our loader functions work in a similar way. They take a DOM element and attach the React Composite to that element. What this means
+is that you can use the UI Composites in any framework that can render HTML elements to the DOM.
+
+**Note:** Something to keep in mind when using these loader functions for the UI Composites is that you are taking on
+not just the bundle for `@azure/communication-react` but also the bundle for `react` and `react-dom`. Best practice here
+is to lazy load these chunks so that you don't get caught waiting for the large bundle when you don't need your
```
Comment (JamesBurnside) [inline packages/storybook8/stories/Composites/TypeScriptLoading/Docs.mdx:28]: This is a tad confusing as we aren't explaining how to lazy load or the fact that internally we have made the react loading async. Could this be reworded more to indicate that react and react-dom uses are set up for lazy loading and using async imports?

File: packages/storybook8/stories/Composites/TypeScriptLoading/Docs.mdx [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+These functions mount each of our React Composites to your applications DOM. The Functions that are available are:
+
+- `loadChatComposite`
+- `loadCallComposite`
+- `loadCallWithChatComposite`
+- `loadOutboundCallComposite`
+
+## How they work
+
+Since all different [Node](https://nodejs.org/en) frameworks are just JavaScript, CSS, and HTML you can use each of the
```
Comment (JamesBurnside) [inline packages/storybook8/stories/Composites/TypeScriptLoading/Docs.mdx:19]: React isn't really a `NodeJs framework`, it's a front-end library. When you bundle tools like redux it start to act like a framework... but even then people argue over the term.
Perhaps we could say something more like
```
Our JavaScript loaders use [react]() under the hood. They handle all of the react complexity and allow the composites to be placed in any application, including applications written with JQuery, Angular, Blazor etc.

They work by taking in a DOM element. They render and attach the React Composite to that element. This means
that you can use the UI Composites in any framework that can render HTML elements to the DOM.

**Note:** Something to keep in mind when using these loader functions .....
```

**Review by JamesBurnside — APPROVED** [review]
Some suggestions, but reads great!

File: packages/storybook8/stories/Composites/TypeScriptLoading/Docs.mdx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+  displayName = 'User';
+  groupId = '<GUID for call>';
+  adapter: any;
+
+  /**
+   * This is how you would load in the CallComposite with lazy loading. This
+   * will only load the CallComposite and the UI library when the function is called.
+   */
+  async loadCalling(): Promise<void> {
+    import('@azure/communication-react/javascript-loaders').then(async (uiLib) => {
```
Comment (PorterNan) [inline packages/storybook8/stories/Composites/TypeScriptLoading/Docs.mdx:99]: these all lines could be just all async/await style for linear logic

const uiLib = await import('@azure/communication-react/javascript-loaders');
this.adapter = await uiLib.loadCallComposiet(...)

this.adapter.on(...)

**Review by PorterNan — APPROVED** [review]

File: packages/storybook8/stories/Composites/TypeScriptLoading/Docs.mdx (hunk 3 above)
Comment (dmceachernmsft) [reply packages/storybook8/stories/Composites/TypeScriptLoading/Docs.mdx:99]: yup totally, do we have a preference here on how we handle the Promise?

**Pull request merged**